linode/terraform-provider-linode
# new: Add `linode_placement_group` resource

## 📝 Description

This pull request adds a new `linode_placement_group` resource that allows users to manage Linode Placement Groups through Terraform.

**NOTE: This pull request does NOT include the logic to manage Placement Group assignment and unassignment. That functionality will be added in TPT-2626 and TPT-2859 as discussed in private.**

## ✔️ How to Test

The following test steps require that you have pulled down this PR locally and have pointed your environment variables at an API instance where placement groups are accessible:

```
export LINODE_URL=...
export LINODE_TOKEN=...
```

### Unit Testing

```
make PKG_NAME=linode/placementgroup unit-test
```

### Integration Testing

```
make PKG_NAME=linode/placementgroup int-test
```

### Manual Testing

1. In a Terraform provider sandbox environment (e.g. dx-devenv), apply the following configuration:

```terraform
resource "linode_placement_group" "group" {
  label = "test"
  region = "us-east"
  affinity_type = "anti_affinity:local"
}
```

2. Ensure the apply finishes successfully.
3. Re-plan the configuration and ensure no changes are planned.
4. Navigate to the new PG in Cloud Manager and ensure all fields match the configuration.
5. Apply the following configuration to update the PG's label:

```terraform
resource "linode_placement_group" "group" {
  label = "test-updated"
  region = "us-east"
  affinity_type = "anti_affinity:local"
}
```

6. Refresh the PG in Cloud Manager and ensure the label has been updated
7. Run `terraform destroy` on the configuration and ensure it completes successfully.
8. Ensure the PG is no longer present in Cloud Manager.
